src/vs/workbench/contrib/terminal/browser/links/terminalWordLinkProvider.ts

```diff
@@ -146,21 +152,43 @@ export class TerminalWordLinkProvider extends TerminalBaseLinkProvider {
 		});
 
 		const sanitizedLink = link.replace(/:\d+(:\d+)?$/, '');
-		const results = await this._searchService.fileSearch(
-			this._fileQueryBuilder.file(this._workspaceContextService.getWorkspace().folders, {
-				// Remove optional :row:col from the link as openEditor supports it
-				filePattern: sanitizedLink,
-				maxResults: 2
-			})
-		);
+		let exactMatch = await this._getExactMatch(sanitizedLink, link);
+		if (exactMatch) {
+			return;
+		}
+		// Fallback to searching quick access
+		return this._quickInputService.quickAccess.show(link);
+	}
```

Comment:
Let's move the detection into the getExactMatch function but keep the handling in activate:

```
const exactMatch = await getExactMatch(...);
if (exactMatch) {
  // handle
  return;
}
// fallback
```